### PR TITLE
libcurl-url.3: overview man page for the URL API

### PR DESCRIPTION
**Pull request by bagder**
Based on the current wiki page: https://github.com/curl/curl/wiki/URL-API

### CODE REVIEW

File: docs/libcurl/libcurl-url.3 [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  rc = curl_url_set(h, CURLUPART_URL, "../test?another", 0);
+.SH "GET URL"
+The `CURLU` handle represents a URL and you can easily extract that with
+\fIcurl_url_get(3)\fP:
+
+  char *url;
+  rc = curl_url_get(h, CURLUPART_URL, &url, 0);
+  curl_free(url);
+
+The zero in the forth argument is a bitmask for changing specific features.
```
Comment (danielgustafsson) [inline docs/libcurl/libcurl-url.3]: s/forth/fourth/

File: docs/libcurl/libcurl-url.3 [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+By "setting" a URL to the handle with \fIcurl_url_set(3)\fP, the URL is parsed
+and stored in the handle. If the URL is not syntactically correct it will
+return an error instead.
+
+.nf
+  rc = curl_url_set(h, CURLUPART_URL,
+                    "https://example.com:449/foo/bar?name=moo", 0);
+.fi
+
+The zero in the forth argument is a bitmask for changing specific features.
```
Comment (danielgustafsson) [inline docs/libcurl/libcurl-url.3]: s/forth/fourth/

File: docs/libcurl/libcurl-url.3 [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  rc = curl_url_get(h, CURLUPART_USER, &user, 0);
+  rc = curl_url_get(h, CURLUPART_PASSWORD, &password, 0);
+  rc = curl_url_get(h, CURLUPART_PORT, &port, 0);
+  rc = curl_url_get(h, CURLUPART_PATH, &path, 0);
+  rc = curl_url_get(h, CURLUPART_QUERY, &query, 0);
+  rc = curl_url_get(h, CURLUPART_FRAGMENT, &fragment, 0);
+.fi
+
+Extracted parts are not URL decoded unless the user also asks for it with the
+CURLU_URLDECODE flag set in the forth bitmask argument.
```
Comment (danielgustafsson) [inline docs/libcurl/libcurl-url.3]: s/fourth/forth/

Comment (danielgustafsson) [inline docs/libcurl/libcurl-url.3]: s/forth/fourth/

Comment (danielgustafsson) [inline docs/libcurl/libcurl-url.3]: s/forth/fourth/

**Comment by danielgustafsson** [comment]
Hmm, the unable to add comment was apparently adding comments after all, and the cancel didn't remove it. Resolved the duplicates.

**Comment by bagder** [comment]
Thanks, pushed a fix.

**Comment by bagder** [comment]
It's a mystery to me how adding a document file can make the test coverage decrease! :-)

**Pull request closed without merging**